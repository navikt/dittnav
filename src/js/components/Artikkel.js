import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage as F } from 'react-intl';

class Artikkel extends Component {
  render() {
    if (!this.props.article) return null;

    const {
      url, preface, lead, heading, className,
    } = this.props.article;

    return (
      <div className="row artikkel">
        <div className="container">
          <div className="col-md-8 col-standalone">
            <article>
              <p className={`preface underline ${className.replace(/_/g, '-')}`}>{preface}</p>
              <h1 className="typo-innholdstittel">{heading}</h1>
              <p>{lead}</p>
              <a data-ga="Dittnav/Artikkel" href={url} className="lenke">
                <F id="dittnav.artikkel.lesmer" />
              </a>
            </article>
          </div>
        </div>
      </div>
    );
  }
}

Artikkel.propTypes = {
  article: PropTypes.shape({
    url: PropTypes.string,
    preface: PropTypes.string,
    lead: PropTypes.string,
    heading: PropTypes.string,
    className: PropTypes.string,
  }),
};

Artikkel.defaultProps = {
  article: null,
};

export default Artikkel;
