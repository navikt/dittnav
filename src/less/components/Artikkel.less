.artikkel {
  margin-left: 0;
  margin-right: 0;
  padding: 30px 30px 50px;
  background: #fff;
  border-top: 1px solid #c0c0c0;

  .preface {
    text-transform: uppercase;
    font-weight: 400;
    font-family: Arial,sans-serif;
    padding-top: 60px;
    margin-bottom: 20px;
    padding-bottom: 20px;
    position: relative;
    text-align: center;
    margin-top: 0;
    font-size: .75rem;
  }

  .lyspaere {
    &::before {
      content: "";
      width: 46px;
      height: 46px;
      background-image: url(../images/lyspaere.svg);
      background-color: transparent;
      background-position: top;
      background-size: 90%;
      display: block;
      background-position: 50%;
      background-repeat: no-repeat;
      left: 50%;
      position: absolute;
      top: 0;
      border-radius: 50%;
      transform: translateX(-50%);
    }
  }

  .typo-innholdstittel {
    text-align: center;
  }
}

@media screen and (max-width: 700px) {
  .artikkel {
    padding: 30px 0px 50px;
  }
}