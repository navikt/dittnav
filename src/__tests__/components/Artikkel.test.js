import * as React from 'react';
import Artikkel from 'js/components/Artikkel';
const ReactTestRenderer = require('react-test-renderer');
import wrapIntl from 'js/Intl';

test('render empty Artikkel', () => {
  const component = ReactTestRenderer.create(wrapIntl(<Artikkel />));
  expect(component.toJSON()).toMatchSnapshot();
});


test('render non-empty Artikkel', () => {
  const article = {
    "url": "https://nav.no/no/Ditt+Nav/DittnavArtikler:443/Friskmelding+til+arbeidsformidling.347278.cms",
    "preface": "Dette er en mindre overskirft",
    "lead": "NAV har en rekke ulike tilretteleggingstiltak. Slike tiltak kan være aktuelle for deg som har en funksjonsnedsettelse, men som kan jobbe så lenge de praktiske forholdene blir lagt til rette. Du kan få hjelp til å vurdere hvordan arbeidsplassen kan tilpasses behovene dine. NAV kan også tilby hjelpemidler, eller gi tilskudd til arbeidsgiver for å gjennomføre nødvendige tilpasninger. Hvis du ser dårlig eller bruker tegnspråk, kan NAV betale en lesehjelp eller en hørselstolk slik at du får den informasjonen du trenger i arbeidshverdagen. NAV kan også betale for transport til arbeidet hvis du ikke kan bruke offentlig transport av medisinske grunner. Noen støtteordninger gjelder bare bedrifter som har avtale om et inkluderende arbeidsliv (IA-virksomheter)",
    "heading": "Friskmelding til arbeidsformidling",
    "className": "lyspaere"
  };

  const component = ReactTestRenderer.create(wrapIntl(<Artikkel article={article} />));
  expect(component.toJSON()).toMatchSnapshot();
});
