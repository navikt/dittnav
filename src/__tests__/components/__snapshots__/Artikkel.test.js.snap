// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render empty Artikkel 1`] = `null`;

exports[`render non-empty Artikkel 1`] = `
<div
  className="row artikkel"
>
  <div
    className="container"
  >
    <div
      className="col-md-8 col-standalone"
    >
      <article>
        <p
          className="preface underline lyspaere"
        >
          Dette er en mindre overskirft
        </p>
        <h1
          className="typo-innholdstittel"
        >
          Friskmelding til arbeidsformidling
        </h1>
        <p>
          NAV har en rekke ulike tilretteleggingstiltak. Slike tiltak kan være aktuelle for deg som har en funksjonsnedsettelse, men som kan jobbe så lenge de praktiske forholdene blir lagt til rette. Du kan få hjelp til å vurdere hvordan arbeidsplassen kan tilpasses behovene dine. NAV kan også tilby hjelpemidler, eller gi tilskudd til arbeidsgiver for å gjennomføre nødvendige tilpasninger. Hvis du ser dårlig eller bruker tegnspråk, kan NAV betale en lesehjelp eller en hørselstolk slik at du får den informasjonen du trenger i arbeidshverdagen. NAV kan også betale for transport til arbeidet hvis du ikke kan bruke offentlig transport av medisinske grunner. Noen støtteordninger gjelder bare bedrifter som har avtale om et inkluderende arbeidsliv (IA-virksomheter)
        </p>
        <a
          className="lenke"
          data-ga="Dittnav/Artikkel"
          href="https://nav.no/no/Ditt+Nav/DittnavArtikler:443/Friskmelding+til+arbeidsformidling.347278.cms"
        >
          <span>
            Les mer om dette
          </span>
        </a>
      </article>
    </div>
  </div>
</div>
`;
